app/models/question.rb

```diff
@@ -35,25 +21,30 @@ class Question < ActiveRecord::Base
   validates :text, presence: true
   validates :number, presence: true, numericality: { only_integer: true }
   validates :question_type, presence: true, inclusion: {in: %w[short_answer multiple_choice true_false]}
-  validates_with QuestionOrdersValidator, on: :next_question
 
   def formatted_question_and_responses
     # TODO (Chris) will implement later since it's pretty complex
     # involves getting the responses too
     text
   end
 
-  # Assumes the response is valid
-  # Assumes the question orders is valid for this question valid?(:next_question) has been checked
+  def is_response_valid?(response)
+    if response_choices.exists?
+      return response_choices.map(&:key).include? response.answer
+    end
+    true
+  end
+
   def next_question(response)
-    if question_type == 'short_answer'
-        return question_orders.first.next_question
-    else
+    if response_choices.exists?
       response_choices.each do |response_choice|
-        if response_choice.key == response.answer
+        if response_choice.question_order.present? && response_choice.key == response.answer
           return response_choice.question_order.next_question
         end
       end
+    elsif question_orders.exists?
+      return question_orders.first.next_question
     end
+    nil
```

Comment:
when i omit this line from the tests, it sometimes returns `nil`. It seems like it depends if it enters the branching logic.